internal/node/node.go

```diff
@@ -193,57 +192,52 @@ func NewBus(cfg BusConfig, dir string, seed types.PrivateKey, logger *zap.Logger
 	chainPath := filepath.Join(consensusDir, "blockchain.db")
 	if _, err := os.Stat(chainPath); os.IsNotExist(err) {
 		if err := sqlStore.ResetChainState(context.Background()); err != nil {
-			return nil, nil, nil, nil, nil, err
+			return nil, nil, nil, err
 		}
 	}
 
 	// create chain database
 	bdb, err := coreutils.OpenBoltChainDB(chainPath)
 	if err != nil {
-		return nil, nil, nil, nil, nil, fmt.Errorf("failed to open chain database: %w", err)
+		return nil, nil, nil, fmt.Errorf("failed to open chain database: %w", err)
 	}
 
 	// create chain manager
 	store, state, err := chain.NewDBStore(bdb, cfg.Network, cfg.Genesis)
 	if err != nil {
-		return nil, nil, nil, nil, nil, err
+		return nil, nil, nil, err
 	}
 	cm := chain.NewManager(store, state)
 
 	// create wallet
 	w, err := wallet.NewSingleAddressWallet(seed, cm, sqlStore, wallet.WithReservationDuration(cfg.UsedUTXOExpiry))
 	if err != nil {
-		return nil, nil, nil, nil, nil, err
-	}
-
-	// create chain subscriber
-	cs, err := chain.NewChainSubscriber(wh, cm, sqlStore, w, time.Duration(cfg.AnnouncementMaxAgeHours)*time.Hour, logger)
-	if err != nil {
-		return nil, nil, nil, nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	// create syncer
 	s, err := NewSyncer(cfg, cm, sqlStore, logger)
 	if err != nil {
-		return nil, nil, nil, nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	b, err := bus.New(ctx, alertsMgr, wh, cm, sqlStore, s, w, sqlStore, sqlStore, sqlStore, sqlStore, sqlStore, sqlStore, logger)
+	// create bus
+	announcementMaxAgeHours := time.Duration(cfg.AnnouncementMaxAgeHours) * time.Hour
+	b, err := bus.New(ctx, alertsMgr, wh, cm, s, w, sqlStore, sqlStore, sqlStore, sqlStore, sqlStore, sqlStore, sqlStore, announcementMaxAgeHours, logger)
```

Comment:
Considering that our goal is to make it easy for someone to create our nodes, how about we create a merged interface like so to be able to only pass the `sqlStore` once instead of 7 times?
```go
type Store interface {
  HostStore // renamed from HostDB
  AutopilotStore
  ChainStore
  MetadataStore
  SettingStore
  EphemeralAccountStore
  MetricsStore
}
```

For unit testing we might need to get a little creative then by adding something like a 
```go
type mockStore {
  hs HostStore
  as AutopilotStore
  ...
}
which allows for mocking individual stores but the improved developer experience when creating a bus might be worth it.

Reply:
Also I think both `worker.New` as well as `bus.New` create `*worker.worker` and `*bus.bus` types which is also not great to have in a library so we should rename `worker` and `bus` to `Worker` and `Bus`.